### ISSUE TITLE
Text shifted down with some characters truncated on Firefox with xtermjs 4.12.0

### ISSUE THREAD

**Issue by dstein64** [xtermjs/xterm.js#3353]
## Details
- Browser and browser version: Firefox 88.0.1
- OS version: Ubuntu 20.10
- xterm.js version: 4.12.0

### Steps to reproduce

Create a page with the following code:

```html
<!doctype html>
<html>
<head>
    <link rel="stylesheet" href="https://unpkg.com/xterm@4.12.0/css/xterm.css" />
    <script src="https://unpkg.com/xterm@4.12.0/lib/xterm.js"></script>
</head>
<body>
<div id="terminal"></div>
<script>
    const config = {
        rendererType: 'canvas',
        fontFamily: 'monospace',
    };
    const term = new Terminal(config);
    term.open(document.getElementById('terminal'));
    term.writeln('Hello world! {Line 1}.');
    term.write('Hello world! {Line 2} $ ');
</script>
</body>
</html>
```

The output has text shifted down (relative to version 4.11.0) with some characters truncated (e.g., see the "{" and "}" characters).

![4_12](https://user-images.githubusercontent.com/541289/120039080-4c5af980-bfd2-11eb-9919-3175ac14fa07.png)

Here's a screenshot using version 4.11.0, where the text is not shifted down nor truncated.

![4_11](https://user-images.githubusercontent.com/541289/120039100-5da40600-bfd2-11eb-95b2-01c633520af7.png)

This issue occurs on Firefox, but not on Chrome.

**Comment by dstein64** [comment]
Running `git bisect` indicates that the issue started with commit a510ffb3c03c2db160579bea6fded2191ee59f0b.

**Comment by dstein64** [comment]
Truncated text can be seen when using GitHub Codespaces on Firefox, but not Chrome.

This might be related to the issue reported here.

However, for Codespaces the truncated text only occurs at the bottom of the screen, as opposed to occurring generally, which suggests it might not be related to the issue here.

For example, here's a screenshot, where the "{}" is only truncated at the bottom.

![truncated_text_at_bottom](https://user-images.githubusercontent.com/541289/120087754-9dddb400-c0b8-11eb-86dc-d53f3ff07494.png)

**Comment by nirui** [comment]
Not sure how this issue is going, but I think `bottom` is a better solution than `ideographic`.

From: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/textBaseline

> "ideographic"
>     The text baseline is the ideographic baseline; this is the bottom of the body of the characters, if the main body of characters protrudes beneath the alphabetic baseline. (Used by Chinese, Japanese, and Korean scripts.)
>
> "bottom"
>     The text baseline is the bottom of the bounding box. This differs from the ideographic baseline in that the ideographic baseline doesn't consider descenders. 

Also, maybe another approach is to make it possible for users to adjust the baseline settings, since `textBaseline` property (along with all the other client text related properties, really) is a mess. Simply lock the option down to one might never be a good idea.

**Comment by Tyriar** [comment]
@nirui I think `bottom` ended up cutting off a bunch of characters as they exceeded the measured dimensions of the cells, see https://github.com/xtermjs/xterm.js/pull/3370#pullrequestreview-705566650

**Comment by dstein64** [comment]
This was resolved in #3370.

I no longer encounter the issue after updating `xtermjs` to `v4.15.0`.
